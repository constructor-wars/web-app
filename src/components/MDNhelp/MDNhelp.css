.searchbar__wrapper {
  display: flex;
  flex-direction: column;
  padding: 10px;
}

.searchbar__bar,
.searchbar__button {
  height: 30px;
  border-radius: 5px;
  background-color: black;
  color: white;
  border: 2px solid white;
}

.searchbar__button {
  margin-top: 10px;
}

.searchbar__bar:hover {
  focus: none;
  border: 2px solid green;
}

.searchbar__button:hover {
  border: 2px solid green;
  color: green;
}

a.search__links__text {
  text-decoration: none;
}

a.search__links__text:hover {
  transform: scale(1.1);
}
