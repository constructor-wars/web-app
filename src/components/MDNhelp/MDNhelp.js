import React from "react";
import "./MDNhelp.css";

export const SearchResultList = ({ data }) =>
  data.map(item => (
    <li className="search__links" key={item.href}>
      <a className="search__links__text" href={item.href} target="_blank">
        {item.textContent}
      </a>
    </li>
  ));

class Searchbar extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      userSearch: "",
      data: []
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.fetchResults = this.fetchResults.bind(this);
  }

  handleChange(event) {
    event.preventDefault();
    this.setState({
      userSearch: event.target.value
    });
  }

  handleSubmit() {
    event.preventDefault();
    this.fetchResults(this.state.userSearch);
  }

  fetchResults(query) {
    const makeHTML = text => {
      const div = document.createElement("div");
      div.innerHTML = text;
      const items = div.querySelectorAll(".result-list-item h4 a");

      this.setState({ data: [] }, () => {
        return [...items].map(item =>
          this.setState(preveState => ({
            data: [item, ...preveState.data]
          }))
        );
      });
    };

    const getResultsFromMdn = q => {
      const url = `https://developer.mozilla.org/en-US/search?q=${q}`;
      fetch(`https://cors-anywhere.herokuapp.com/${url}`)
        .then(res => res.text())
        .then(text => makeHTML(text))
        .catch(err => console.log(err));
    };

    getResultsFromMdn(query);
  }

  render() {
    return (
      <div className="searchbar__wrapper">
        <input
          className="searchbar__bar"
          placeholder="Feeling stuck..? Let MDN help..."
          value={this.state.userSearch}
          onChange={this.handleChange}
        />
        <button
          className="searchbar__button"
          type="Submit"
          onClick={this.handleSubmit}
        >
          Submit
        </button>
        <ul>
          <SearchResultList data={this.state.data} />
        </ul>
      </div>
    );
  }
}

export default Searchbar;
