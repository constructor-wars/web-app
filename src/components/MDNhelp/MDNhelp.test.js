import React from "react";
import MDNhelp, { SearchResultList } from "./MDNhelp";
import renderer from "react-test-renderer";

describe("MDNhelp", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<MDNhelp />).toJSON();
    expect(tree).toMatchInlineSnapshot(`
<div
  className="searchbar__wrapper"
>
  <input
    className="searchbar__bar"
    onChange={[Function]}
    placeholder="Feeling stuck..? Let MDN help..."
    value=""
  />
  <button
    className="searchbar__button"
    onClick={[Function]}
    type="Submit"
  >
    Submit
  </button>
  <ul />
</div>
`);
  });

  it("matches the snapshot", () => {
    const data = [
      { textContent: "one", href: "http://mdn.com/1" },
      { textContent: "two", href: "http://mnd.com/2" }
    ];
    const tree = renderer.create(<SearchResultList data={data} />).toJSON();
    expect(tree).toMatchInlineSnapshot(`
Array [
  <li
    className="search__links"
  >
    <a
      className="search__links__text"
      href="http://mdn.com/1"
      target="_blank"
    >
      one
    </a>
  </li>,
  <li
    className="search__links"
  >
    <a
      className="search__links__text"
      href="http://mnd.com/2"
      target="_blank"
    >
      two
    </a>
  </li>,
]
`);
  });
});
