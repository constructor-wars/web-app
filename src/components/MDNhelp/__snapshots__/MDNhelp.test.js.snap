// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MDNhelp matches the snapshot 1`] = `
<div>
  <input
    className="searchbar"
    onChange={[Function]}
    placeholder="Feeling stuck..? Let MDN help..."
    value=""
  />
  <button
    onClick={[Function]}
    type="Submit"
  >
    Submit
  </button>
  <ul />
</div>
`;
